x-pack/plugins/observability_solution/synthetics/public/apps/synthetics/components/alerts/tls_rule_ui.tsx

```diff
@@ -23,9 +23,12 @@ export const TLSRuleComponent: React.FC<{
   const { settings } = useSelector(selectDynamicSettings);
 
   useEffect(() => {
-    if (typeof settings === 'undefined') {
-      dispatch(getDynamicSettings());
-    }
+    const dispatchDynamicSettings = async () => {
+      if (typeof settings === 'undefined') {
+        dispatch(setDynamicSettingsAction.get(await getDynamicSettings()));
+      }
+    };
+    dispatchDynamicSettings();
```

Comment:
@shahzad31 @dominiqueclarke I attempted to fix the error, now it shows two toasts as seen in the below screenshot. Is this a desired behavior?
Also, what does the dynamic settings do when opening new rule flyout?

<img width="599" alt="Screenshot 2024-11-30 at 14 56 58" src="https://github.com/user-attachments/assets/a64c2598-4b39-4634-94c1-5e4b3a4fade2">